src/app/content/components/BookBanner.tsx

```diff
@@ -176,158 +176,137 @@ export const BarWrapper = styled.div<BarWrapperProps>`
-              <LeftArrow colorSchema={book.theme} />{book.tree.title}
+              <LeftArrow colorSchema={bookWithTheme.theme} />{bookWithTheme.tree.title}
             </BookTitleLink>
         }
         {treeSection
           ? <BookChapter
-            colorSchema={book.theme}
-            dangerouslySetInnerHTML={{__html: treeSection.title}}
+            colorSchema={bookWithTheme.theme}
+            dangerouslySetInnerHTML={{ __html: treeSection.title }}
           />
           : null}
       </TopBar>
     </BarWrapper>,
     <BarWrapper
-      colorSchema={book.theme}
+      colorSchema={bookWithTheme.theme}
       variant='mini'
       key='mini-nav'
-      ref={this.miniBanner}
+      ref={miniBanner}
       data-testid='bookbanner-collapsed'
       data-analytics-region='book-banner-collapsed'
     >
       <TopBar>
         {
-          bookUrl === undefined
-            ? <BookTitle data-testid='book-title-collapsed' colorSchema={book.theme} variant='mini'>
-              {book.tree.title}
+          url === undefined
+            ? <BookTitle data-testid='book-title-collapsed' colorSchema={bookWithTheme.theme} variant='mini'>
+              {bookWithTheme.tree.title}
             </BookTitle>
             : <BookTitleLink
               data-testid='details-link-collapsed'
-              href={bookUrl}
+              href={url}
               variant='mini'
-              colorSchema={book.theme}
+              colorSchema={bookWithTheme.theme}
               onClick={(e: React.MouseEvent<HTMLAnchorElement>) => {
-                this.handleLinkClick(e, bookUrl);
+                handleLinkClick(e, url);
               }}
-              tabIndex={this.state.tabbableBanner === 'mini' ? undefined : -1}
+              tabIndex={tabbableBanner === 'mini' ? undefined : -1}
             >
-            <LeftArrow colorSchema={book.theme} />{book.tree.title}
-          </BookTitleLink>
+              <LeftArrow colorSchema={bookWithTheme.theme} />{bookWithTheme.tree.title}
+            </BookTitleLink>
         }
         {treeSection
           ? <BookChapter
-            colorSchema={book.theme}
+            colorSchema={bookWithTheme.theme}
             variant='mini'
-            dangerouslySetInnerHTML={{__html: treeSection.title}}
+            dangerouslySetInnerHTML={{ __html: treeSection.title }}
           />
           : null
         }
       </TopBar>
     </BarWrapper>,
   ]);
-}
 
-export default connect(
-  (state: AppState) => ({
-    book: select.book(state),
-    bookTheme: select.bookTheme(state),
-    hasUnsavedHighlight: hasUnsavedHighlight(state),
-    pageNode: select.pageNode(state),
-  })
-)(BookBanner);
+  if (!book) {
+    return <BarWrapper colorSchema={undefined} up={false} />;
+  }
+
+  const bookUrl = hasOSWebData(book)
+    ? book.book_state !== 'retired'
+      ? bookDetailsUrl(book)
+      : undefined
+    : undefined;
+
+  return <>{renderBars({theme: bookTheme, ...book}, bookUrl, pageNode)}</>;
```

Comment:
i'm not really feeling like this function is doing much right? might as well move the stuff all to just be down here